pkg/ingester/transfer.go

```diff
@@ -35,63 +35,79 @@ func (i *Ingester) TransferChunks(stream client.Ingester_TransferChunksServer) e
 	seriesReceived := 0
 	xfer := func() error {
 		userStates := newUserStates(i.limiter, i.cfg, i.metrics)
+		var chunksAdded float64
+		// run as an anonymous function so only a single error needs handling if returned
+		err := func() error {
+			for {
+				wireSeries, err := stream.Recv()
+				if err == io.EOF {
+					break
```

Comment:
Can we use `defer` here to handle failure instead of putting everything into new anonymous function?

Reply:
I personally find error handling in defer a little too magical. I'd rather be explicit.

Reply:
Yes, it is. But having two nested anonymous functions is just too ugly. Can we move it to named function at least, or perhaps a method?